install/operator/pkg/syndesis/action/install.go

```diff
@@ -107,110 +136,355 @@ func (a *installAction) Execute(ctx context.Context, syndesis *v1alpha1.Syndesis
 
-			operation.SetNamespaceAndOwnerReference(addon, syndesis)
+	// It might be already Unstructured..
+	if u, ok := obj.(*unstructured.Unstructured); ok {
+		return u, nil
+	}
 
-			err = createOrReplace(ctx, a.client, addon)
-			if err != nil && !k8serrors.IsAlreadyExists(err) {
+	// Convert it..
+	fields, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
+	if err != nil {
+		return nil, err
+	}
+	return &unstructured.Unstructured{fields}, nil
+}
 
-				if meta.IsNoMatchError(err) {
-					// CRD for that resource type is not available.
-					a.log.Info("Skipping install resource.  CRD not installed?", "GroupVersionKind", addon.GroupVersionKind(), "Name", addon.GetName())
-				} else {
-					if bytes, err := yaml.Marshal(addon); err == nil {
-						fmt.Printf("Failed to create or replace resource:\n%s\n", string(bytes))
+func CreateOrUpdate(ctx context.Context, cl client.Client, o runtime.Object, skipFields ...string) (*unstructured.Unstructured, controllerutil.OperationResult, error) {
+	if len(skipFields) == 0 {
+		skipFields = append(skipFields, "kind", "apiVersion", "status")
+	}
+	desired, err := ToUnstructured(o)
+	if err != nil {
+		return desired, controllerutil.OperationResultNone, err
+	}
+
+	createdCopy := desired.DeepCopy()
+	modType, err := controllerutil.CreateOrUpdate(ctx, cl, createdCopy, func(o runtime.Object) error {
+		existing := o.(*unstructured.Unstructured)
+
+		if desired.GetAPIVersion() == "v1" && desired.GetKind() == "Secret" {
+			if desired.Object["stringData"] != nil && existing.Object["data"] != nil {
+				from := desired.Object["stringData"].(map[string]interface{})
+				to := existing.Object["data"].(map[string]interface{})
+				updates := map[string]interface{}{}
+				for key, value := range from {
+					plain := value.(string)
+					encoded := base64.StdEncoding.EncodeToString([]byte(plain))
+					if to[key] != encoded {
+						updates[key] = value
 					}
-					return err
+				}
+				if len(updates) > 0 {
+					desired.Object["stringData"] = updates
+				} else {
+					delete(desired.Object, "stringData")
 				}
 			}
 		}
+
+		mergePath := desired.GetAPIVersion() + "/" + desired.GetKind()
+
+		for key, value := range desired.Object {
+			if swag.ContainsStrings(skipFields, key) {
+				continue
+			}
+			switch key {
+			case "metadata":
+				to := existing.Object[key].(map[string]interface{})
+				from := value.(map[string]interface{})
+				mergeMap(mergePath+"/metadata", to, from)
+			case "spec": // skip updating this field.
+				to := existing.Object[key].(map[string]interface{})
+				from := value.(map[string]interface{})
+				mergeMap(mergePath+"/metadata", to, from)
+			default:
+				existing.Object[key] = value
+			}
+		}
+
+		//if d.GetKind() == "DeploymentConfig" && d.GetName() == "syndesis-meta" {
+		//	Debug("existing:", existing, "(index .Object.spec.template.spec.containers 0).resources.limits.memory")
+		//}
```

Comment:
Do you still want to keep this commented-out code?

Reply:
It comes in handy when you need to debug stuff.  It's easy to get into a situation where the same resource keeps getting updated by the operator for no good reason.   I've run in to a couple of case of that: For example the the limit you put into the template, returned with a different type of unit, and so that causes a update.